### ISSUE TITLE
ReferenceError: window is not defined

### ISSUE THREAD

**Issue by cecilehautecoeur** [stephanrauh/ngx-extended-pdf-viewer#1719]
Hello,

**Describe the bug**
The following warning appears when displaying some documents:
Warning: getOperatorList - ignoring XObject: "ReferenceError: window is not defined".

In that case, most elements of the pdf are missing (e.g. the text and some images).

**Version info**
Version 16.X.X
The bug seems not to happen in previous versions. We tested on version 15.

**Desktop (please complete the following information):**
Chrome. We also tested on Safari and the bug does not seem to appear there.

It seems that the bug has already been fixed in version 11.0.0-alpha.12 #1059.

Thank you

**Comment by stephanrauh** [comment]
The message `"getOperatorList - ignoring XObject"` only occurs in the JavaScript sandbox. It doesn't show in any of my test files. Can you send me a PDF file causing this warning, please?

**Comment by RodolfoGraziano** [comment]
Hi, i'd like to add that i get the same error with a single specific file.

[Warning: getOperatorList - ignoring XObject: "ReferenceError: window is not defined".](pdf.worker-3.4.493.min.js:22 Warning: getOperatorList - ignoring XObject: "ReferenceError: window is not defined".)

Unfortunately I can't share the file since it has confidential info inside. 
However, id' like to add that the bug does not seem to appear when the file is loaded inside the demo at https://pdfviewer.net/extended-pdf-viewer/getting-started.
It is present however in a clean new project that follows the minimal setup step for the ngx-extended-pdf-viewer (npm add follower by the asset glob setup in the angular.json) made in Angular 15.1.0.
Is there some specific configuration that are present in the demo that are not reported in the generic setup steps?

**Comment by stephanrauh** [comment]
No, I don't think there's any specific configuration. You can look it up yourself: https://github.com/stephanrauh/ngx-extended-pdf-viewer/tree/main/projects/ngx-extended-pdf-viewer/schematics/ng-add. All the magic of `ng add ngx-extended-pdf-viewer` sits in the `index.ts`, which is a pretty short file, so you can browse it quickly.

**Comment by sudilshr** [comment]
Hi @stephanrauh,

I am facing the same issue.

Angular: 15.2.0
ngx-extended-pdf-viewer: 17.0.0-alpha.0
pdfjs version: 3.5.528

`window.pdfDefaultOptions.activateWillReadFrequentlyFlag` in pdf.worker file might be the source of the problem.
https://github.com/stephanrauh/ngx-extended-pdf-viewer/blob/95bc342c32805e746fb7187253ea33752ac00b51/projects/ngx-extended-pdf-viewer/assets/pdf.worker-3.5.528.js#L36025

`createMask` and `FakeUnicodeFont` is checking `window.pdfDefaultOptions.activateWillReadFrequentlyFlag`. The PDF I have issue with has a `/ImageMask` in it.

> << /Type /XObject /Subtype /Image /Width 2176 /Height 2972 /ImageMask true
> /Interpolate true /BitsPerComponent 1 /Length 113993 /Filter /FlateDecode >>


I am not allowed to share the original file so I generated a sample PDF with `/ImageMask` to replicate the issue.
https://github.com/sudilshr/pdf-viewer-masked-image-issue/blob/bf99814abf2c94572e918e04b98be17252ae171a/src/assets/sandwich-masked.pdf

https://stackblitz.com/edit/github-uf3tfg

**Comment by michaelmarcuccio** [comment]
This issue is happening to us too, it completely jumbles/breaks (potentially without the user being able to realize it) the rendering of the PDFs. I tried the flag you mentioned above with true and false. Neither one helps fix the issue.

I have isolated the rendering issue to starting in v16.2.4 as it does not occur in v16.2.3. We have to downgrade back to v16.2.3 to avoid the issue. I would consider this issue a major bug as it silently fails to render PDFs that would previously be rendered by this library. Anyone upgrading can run into this issue and not even realize it.

**Comment by sudilshr** [comment]
@stephanrauh - I was not aware Stackblitz url I shared was doing server side rendering. My original project is browser side rendering though.

I log the original error on the console and the error is originating from `PDFImage.createMask` line 36027 which is `const options = window.pdfDefaultOptions.activateWillReadFrequentlyFlag ? {`.

> PDF bd650c3e558bb8f69adc5bc1ce139a10 [1.6 GPL Ghostscript 9.21 / ocrmypdf 5.3.3 / Tesseract OCR-PDF 3.05.01] (PDF.js: 3.5.528 [97fe1fa13])  modified by ngx-extended-pdf-viewer
> ReferenceError: window is not defined
>     at PDFImage.createMask (pdf.worker-3.5.528.min.js:36027:23)
>     at PartialEvaluator.buildPaintImageXObject (pdf.worker-3.5.528.min.js:8796:39)
>     at pdf.worker-3.5.528.min.js:9577:22
>     at new Promise (<anonymous>)
>     at promiseBody (pdf.worker-3.5.528.min.js:9540:18)
>     at pdf.worker-3.5.528.min.js:9510:13
> pdf.worker-3.5.528.min.js:22 Warning: getOperatorList - ignoring XObject: "ReferenceError: window is not defined".

AFAIK `window` is not accessible inside web worker. I am guessing that's the problem.

**Comment by stephanrauh** [comment]
Right you are! During the last couple of week, I worked a lot to support server-side rendering, so I missed the real cause of the issue. It's good that you, @sudilshr, mentioned the web worker. I've introduced the bug accidentally when implementing #1659. And kudos for providing me with a test PDF. I didn't have a file to test this particular feature yet, so your file comes in handy!

Your bugfix has landed with version 17.0.0-alpha.1.

Enjoy!
Stephan
